// Copyright 2025 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//
syntax = "proto2";

package test_namespace;

message FooInput {
  optional float a = 1;
  optional string string_field = 3;
  optional float unused_field = 4;

  message BarInput {
    optional float c = 1;
    repeated FooInput nested_foo = 2;
    optional float unused_field = 3;

    extensions 100000 to 199999;
  }
}

message Output {
  optional float result = 1;
  optional float unused_field = 2;

  extensions 100000 to 199999;
}
